### PR TITLE
fix #239 , use python-style templating instead of Jinja in the plugins login_cmd_template

### LINKED ISSUE #239: molecule login results in "Bad port '{port}'"
Using Molecule on MacOS ventura 13.6.4

```
molecule 24.2.0 using python 3.11
    ansible:2.16.3
    azure:23.5.3 from molecule_plugins
    containers:23.5.3 from molecule_plugins requiring collections: ansible.posix>=1.3.0 community.docker>=1.9.1 containers.podman>=1.8.1
    default:24.2.0 from molecule
    docker:23.5.3 from molecule_plugins requiring collections: community.docker>=3.4.11 ansible.posix>=1.4.0
    ec2:23.5.3 from molecule_plugins
    gce:23.5.3 from molecule_plugins requiring collections: google.cloud>=1.0.2 community.crypto>=1.8.0
    openstack:23.5.3 from molecule_plugins requiring collections: openstack.cloud>=2.1.0
    podman:23.5.3 from molecule_plugins requiring collections: containers.podman>=1.7.0 ansible.posix>=1.3.0
    vagrant:23.5.3 from molecule_plugins
```

This seems to be a new issue for me, but when using molecule to test against a remote EC2 instance, when I run "molecule login -s my_scenario", I get `Bad port '{port}'`

Molecule converge seems to work fine.

Scenario specific configuration:

ansible_inventory.yml

```yaml
# Molecule managed
---
all:
  hosts:
    aws-marketplace-az2-ec2-stig-localtest: &id001
      ansible_host: x.x.x.x
      ansible_port: '22'
      ansible_private_key_file: /Users/isuftin/.cache/molecule/stig/aws-marketplace-az2-ec2/id_rsa
       ansible_ssh_common_args: -o UserKnownHostsFile=/dev/null -o ControlMaster=auto
         -o ControlPersist=60s -o ForwardX11=no -o LogLevel=ERROR -o IdentitiesOnly=yes
         -o StrictHostKeyChecking=no
       ansible_user: ec2-user
       connection: ssh
   vars: &id002
     molecule_ephemeral_directory: '{{ lookup(''env'', ''MOLECULE_EPHEMERAL_DIRECTORY'')
       }}'
     molecule_file: '{{ lookup(''env'', ''MOLECULE_FILE'') }}'
     molecule_instance_config: '{{ lookup(''env'', ''MOLECULE_INSTANCE_CONFIG'') }}'
     molecule_no_log: '{{ lookup(''env'', ''MOLECULE_NO_LOG'') or not molecule_yml.provisioner.log|default(False)
       | bool }}'
     molecule_scenario_directory: '{{ lookup(''env'', ''MOLECULE_SCENARIO_DIRECTORY'')
       }}'
     molecule_yml: '{{ lookup(''file'', molecule_file) | from_yaml }}'
 az2:
   hosts:
     aws-marketplace-az2-ec2-stig-localtest: *id001
   vars: *id002
 ec2:
   hosts:
     aws-marketplace-az2-ec2-stig-localtest:
       ansible_host: x.x.x.x
       ansible_port: '22'
       ansible_private_key_file: /Users/isuftin/.cache/molecule/stig/aws-marketplace-az2-ec2/id_rsa
       ansible_ssh_common_args: -o UserKnownHostsFile=/dev/null -o ControlMaster=auto
         -o ControlPersist=60s -o ForwardX11=no -o LogLevel=ERROR -o IdentitiesOnly=yes
         -o StrictHostKeyChecking=no
       ansible_user: ec2-user
       connection: ssh
   vars:
     molecule_ephemeral_directory: '{{ lookup(''env'', ''MOLECULE_EPHEMERAL_DIRECTORY'')
       }}'
     molecule_file: '{{ lookup(''env'', ''MOLECULE_FILE'') }}'
     molecule_instance_config: '{{ lookup(''env'', ''MOLECULE_INSTANCE_CONFIG'') }}'
     molecule_no_log: '{{ lookup(''env'', ''MOLECULE_NO_LOG'') or not molecule_yml.provisioner.log|default(False)
       | bool }}'
     molecule_scenario_directory: '{{ lookup(''env'', ''MOLECULE_SCENARIO_DIRECTORY'')
       }}'
     molecule_yml: '{{ lookup(''file'', molecule_file) | from_yaml }}'
 ungrouped:
   vars: {}
```

ansible.cfg
```ini
# Molecule managed

[defaults]
ansible_managed = Ansible managed: Do NOT edit this file manually!
display_failed_stderr = True
forks = 50
retry_files_enabled = False
host_key_checking = False
nocows = 1
interpreter_python = auto_silent
bin_ansible_callbacks = True
callbacks_enabled = ansible.posix.profile_tasks,ansible.posix.profile_roles,ansible.posix.timer,ansible.builtin.junit
display_args_to_stdout = True
pipelining = True
show_custom_stats = True
stdout_callback = community.general.yaml
timeout = 60
use_persistent_connections = True
verbosity = 2
fact_caching = jsonfile
fact_caching_connection = /Users/isuftin/.cache/molecule/stig/aws-marketplace-az2-ec2/facts_cache
fact_caching_timeout = 7200
[ssh_connection]
scp_if_ssh = True
control_path = %(directory)s/%%h-%%p-%%r
[diff]
always = True
[galaxy]
display_progress = False
[ssh]
ssh_args = -C -o ControlMaster=auto -o ControlPersist=600s -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o ConnectTimeout=60
reconnection_retries = 3
```

instance_config.yml
```yaml
- address: x.x.x.x (aws internal ip addr)
  identity_file: /Users/isuftin/.cache/molecule/stig/aws-marketplace-az2-ec2/id_rsa
  instance: aws-marketplace-az2-ec2-stig-localtest
  instance_ids: [i-my-ec2-id]
  port: '22'
  user: ec2-user
```

molecule.yml

```yaml
# Molecule managed

---
dependency:
  command: null
  enabled: true
  env: {}
  name: galaxy
  options:
    force: true
    requirements-file: molecule/common/collections_requirements.yml
    role-file: molecule/common/role_requirements.yml
driver:
  name: ec2
  options:
    managed: true
  provider:
    name: null
  safe_files: []
  ssh_connection_options: []
platforms:
  - aws_iam_role: my_iam_role
    groups:
      - ec2
      - az2
    image: ami-my=ami
    instance_type: t3a.medium
    key_inject_method: ec2
    name: aws-marketplace-az2-ec2-stig-localtest
    region: us-east-1
    security_groups: sg-my-security-groups
    ssh_startup_wait_seconds: 60
    tags:
      Name: Ansible Molecule - aws-marketplace-az2-ec2-stig-localtest
      wma:ansible_managed: 'true'
      wma:molecule_pipeline_id: manual
      wma:project_id: enterprise
    volume_name: /dev/xvda
    vpc_id: vpc-my-vpc
    vpc_subnet_id: subnet-my-subnet
prerun: true
provisioner:
  ansible_args: []
  children: {}
  config_options:
    defaults:
      bin_ansible_callbacks: true
      callbacks_enabled: ansible.posix.profile_tasks,ansible.posix.profile_roles,ansible.posix.timer,ansible.builtin.junit
      display_args_to_stdout: true
      fact_caching: jsonfile
      fact_caching_connection: /Users/isuftin/.cache/molecule/stig/aws-marketplace-az2-ec2/facts_cache
      fact_caching_timeout: 7200
      pipelining: true
      show_custom_stats: true
      stdout_callback: community.general.yaml
      timeout: 60
      use_persistent_connections: true
      verbosity: 2
    diff:
      always: true
    galaxy:
      display_progress: false
    ssh:
      reconnection_retries: 3
      ssh_args: -C -o ControlMaster=auto -o ControlPersist=600s -o IdentitiesOnly=yes
        -o PreferredAuthentications=publickey -o ConnectTimeout=60
  connection_options: {}
  env:
    SSH_USERNAME: ec2-user
  inventory:
    group_vars: {}
    host_vars: {}
    hosts: {}
    links:
      group_vars: ../common/group_vars
      host_vars: ../common/ec2/host_vars
  log: true
  name: ansible
  options: {}
  playbooks:
    cleanup: cleanup.yml
    converge: ../common/converge.yml
    create: ../common/ec2/create.yml
    destroy: ../common/ec2/destroy.yml
    prepare: prepare.yml
    side_effect: side_effect.yml
    verify: verify.yml
role_name_check: 0
scenario:
  check_sequence:
    - dependency
    - cleanup
    - destroy
    - create
    - prepare
    - converge
    - check
    - cleanup
    - destroy
  cleanup_sequence:
    - cleanup
  converge_sequence:
    - dependency
    - create
    - prepare
    - converge
  create_sequence:
    - dependency
    - create
    - prepare
  destroy_sequence:
    - dependency
    - cleanup
    - destroy
  name: aws-marketplace-az2-ec2
  test_sequence:
    - dependency
    - cleanup
    - destroy
    - syntax
    - create
    - prepare
    - converge
    - idempotence
    - side_effect
    - verify
    - cleanup
    - destroy
verifier:
  additional_files_or_dirs: []
  directory: ../common/tests
  enabled: true
  env: {}
  name: testinfra
  options:
    instafail: true
    junit-xml: junit.xml
    maxprocesses: '10'
    n: auto
    r: fEs
    v: true
```

```sh
 molecule login -s aws-marketplace-az2-ec2
INFO     Found config file /ansible/roles/stig/.config/molecule/config.yml
WARNING  Driver vagrant does not provide a schema.
WARNING  Driver vagrant does not provide a schema.
WARNING  Driver ec2 does not provide a schema.
WARNING  Driver ec2 does not provide a schema.
WARNING  Driver ec2 does not provide a schema.
WARNING  Driver ec2 does not provide a schema.
WARNING  Driver ec2 does not provide a schema.
INFO     Running aws-marketplace-az2-ec2 > login
Bad port '{port}'
```

I can use the ssh key defined in the inventory along with the ip address and SSH into the instance manually with no issues.

### PR DESCRIPTION
**Pull request by danielpodwysocki**
This file assumes Jinja templating for the port parameter and passes a Jinja-style "{{ port }}".

https://github.com/ansible/molecule/blob/main/src/molecule/command/login.py#L105

When it reaches molecule, it is subsituted in  this file and is done by python calling `.format()` on the string. That causes it to not render correctly and gives users issues running molecule login.

ref: https://github.com/ansible-community/molecule-plugins/issues/239

### CODE REVIEW

**Comment by danielpodwysocki** [comment]
I tested by editing this change into my venv locally and can now log in correctly to ec2 instances by calling `molecule login`

**Comment by danielpodwysocki** [comment]
It affects users on all plugins - updated the PR to address each.

**Comment by danielpodwysocki** [comment]
Any news on this?

**Comment by danielpodwysocki** [comment]
Any news on this PR?

This bug breaks some of the plugin functionality quite badly.

**Review by cedricverhaeghe — APPROVED** [review]
I applied these changed locally and can confirm it fixes the issue.

**Comment by ekin-oener-provsn** [comment]
Can also confirm, that this fixes the issue.

**Review by ekin-oener-provsn — APPROVED** [review]

**Comment by ekin-oener-provsn** [comment]
@danielpodwysocki Please add a label to the PR. Would recommend to use 'patch' here. Otherwise following check is failing to success 'ack / ack / ack (pull_request_target) '.
I think, after adding this label, the process of merging will be a step nearer.

**Comment by danielpodwysocki** [comment]
> @danielpodwysocki Please add a label to the PR. Would recommend to use 'patch' here. Otherwise following check is failing to success 'ack / ack / ack (pull_request_target) '. I think, after adding this label, the process of merging will be a step nearer.

@ekin-oener-provsn I miss permissions to do that here, I could only create the PR but seems I cannot edit labels. It will need to be someone with enough access to this repo.

File: src/molecule_plugins/gce/driver.py
Comment (mikerosile) [inline src/molecule_plugins/gce/driver.py:99]: Looks like there's an extra closing curly brace for line 99, `-p {port}}`, which should read `-p {port}`

Comment (danielpodwysocki) [reply src/molecule_plugins/gce/driver.py]: Good catch - Fixed that up.

**Comment by dzintars** [comment]
Confirm. Fixes the same issue for me.

**Comment by StarlessNights** [comment]
Fixes the issue on Arch Linux. Seems good to merge.

**Comment by cclose** [comment]
Bump... the fix seems good, we just need a maintainer to add the label to the PR as the submitter asserts they lack permission to edit the labels.

**Comment by robpou** [comment]
Bump. Please merge this, the bug is breaking stuff.

**Comment by ben16w** [comment]
Can confirm the PR fixed the issue locally for me, is it possible for it to be merged?

**Comment by matt-horwood-mayden** [comment]
please please please get this merged, as we are re-writing all our ansible roles and this would help so much

**Comment by isaacfraser-tomtom** [comment]
> please please please get this merged, as we are re-writing all our ansible roles and this would help so much

+1 please approve this.

**Comment by matt-horwood-mayden** [comment]
@audgirka are you able to progress this PR please?

**Review by apatard — APPROVED** [review]

**Comment by matt-horwood-mayden** [comment]
@apatard Have had a quick look at the failed builds and can see that `centos:7` is failing, as thats now EOL should that be removed?

**Comment by apatard** [comment]
> @apatard Have had a quick look at the failed builds and can see that `centos:7` is failing, as thats now EOL should that be removed?

I found time to fix that the other day but didn't find enough time to clean, push and create a PR. I hope I'll do it this week

**Comment by apatard** [comment]
PR opened, and looks like the CI is green, so only a reviewer is needed to get it merged. This should unblock this PR

**Comment by danielpodwysocki** [comment]
Any news here?

This is a rather annoying bug to work around and feels like something we should be addressing in weeks, not months?

**Comment by apatard** [comment]
> Any news here?
> 
> This is a rather annoying bug to work around and feels like something we should be addressing in weeks, not months?

once the CI is fixed, this PR will be merged. As long as my PR about fixing the CI is not merged (in this form or any other form), I fear it's stuck :(

**Comment by apatard** [comment]
Right, looks like my PR has been merged (big thanks to @ssbarnea). Once this PR is rebased, it'll be ready to be merged I guess.

**Comment by danielpodwysocki** [comment]
> Right, looks like my PR has been merged (big thanks to @ssbarnea). Once this PR is rebased, it'll be ready to be merged I guess.

Awesome - I've rebased off of `main`.

**Comment by simi** [comment]
Any chance to finally release? :pray:

**Pull request merged**
